src/v/cloud_topics/level_one/tests/simple_metastore_test.cc

```diff
@@ -17,9 +17,9 @@ using namespace experimental::cloud_topics::l1;
 
 namespace {
 
-const object_id oid1{uuid_t::create()};
-const object_id oid2{uuid_t::create()};
-const object_id oid3{uuid_t::create()};
+const object_id oid1{.epoch = cluster_epoch{1}, .name = uuid_t::create()};
+const object_id oid2{.epoch = cluster_epoch{1}, .name = uuid_t::create()};
+const object_id oid3{.epoch = cluster_epoch{1}, .name = uuid_t::create()};
```

Comment:
Curious why you chose to bake epoch into the object ID, vs leaving it separate or having a separate `l0::object_id{epoch, object_id}` or somesuch. It's a little surprising, given we don't have the concept of epochs in L1.

Reply:
I guess the code that generates the actual object names for uploads can just ignore the epoch. But we probably need some bit-flag in the object_id that will tell us if it's L0 or L1. It can be a separate type but in this case the `extent_meta` should include the variant that can be either L0 or L1 object id.

Reply:
My kneejerk reaction is that we shouldn't necessarily try to fit `extent_meta` to both L0 and L1, given they are separate formats. If it works out such that we end up sharing a lot of code, then great. But it doesn't feel like we need to design for that from day one.